src/ts/services/keyring-vault.ts

```diff
@@ -0,0 +1,171 @@
+    this._mnemonic = undefined
+    this._keyring = undefined
+  }
+
+  generateMnemonic (): string {
+    if (this.isLocked()) {
+      throw new Error(t('error.wallet.locked'))
+    }
+    this._mnemonic = mnemonicGenerate()
+    return this._mnemonic
+  }
+
+  createAccount (mnemonic: string, accountName: string): Promise<KeyringPair$Json> {
+    if (this.isLocked()) throw new Error(t('error.wallet.locked'))
+    if (this._mnemonic !== mnemonic) throw new Error(t('error.mnemonic.unmatched'))
+    let pair = this.keyring.addFromMnemonic(mnemonic, { name: accountName })
+    this._mnemonic = undefined
+    return this.saveAccount(pair).then((keyringPair$Json: KeyringPair$Json) => {
+      return keyringPair$Json
+    })
+  }
+
+  updateAccountName (address: string, accountName: string): Promise<KeyringPair$Json> {
+    if (this.isLocked()) throw new Error(t('error.wallet.locked'))
+    const pair = this.keyring.getPair(address)
+    if (!pair) throw new Error(t('error.account.notFound'))
+    let meta = pair.getMeta()
+    meta.name = accountName
+    pair.setMeta(meta)
+    this.keyring.addPair(pair)
+    return this.saveAccount(pair).then((json) => {
+      return json
+    })
+  }
+
+  removeAccount (address: string) {
+    if (this.isLocked()) throw new Error(t('error.wallet.locked'))
+    this.keyring.removePair(address)
+    LocalStore.get(this._vaultKey).then((vault) => {
+      if (vault && vault[this._vaultKey]) {
+        delete vault[this._vaultKey][address]
+        return LocalStore.set(vault[this._vaultKey])
+      }
+      return
+    })
+  }
+
+  isMnemonicValid (mnemonic: string): boolean {
+    return mnemonicValidate(mnemonic)
+  }
+
+  importAccountFromMnemonic (mnemonic: string, accountName: string): Promise<KeyringPair$Json> {
+    if (this.isLocked()) throw new Error(t('error.wallet.locked'))
+    if (!this.isMnemonicValid(mnemonic)) throw new Error(t('error.mnemonic.invalid'))
+    let pair = this.keyring.createFromUri(mnemonic, { name: accountName, imported: true })
+    this.keyring.addPair(pair)
+    return this.saveAccount(pair).then((keyringPair$Json: KeyringPair$Json) => {
+      return keyringPair$Json
+    })
+  }
+
+  importAccountFromJson (json: KeyringPair$Json, password?: string): Promise<KeyringPair$Json> {
+    if (this.isLocked()) throw new Error(t('error.wallet.locked'))
+    let pair = this.keyring.addFromJson(json)
+    let meta = pair.getMeta()
+    meta.imported = true
+    pair.setMeta(meta)
+    if (password) {
+      pair.decodePkcs8(password)
+    }
+    this.keyring.addPair(pair)
+    return this.saveAccount(pair).then((json) => {
+      return json
+    })
+  }
+
+  isLocked (): boolean {
+    return !this._password
+  }
+
+  isUnlocked (): boolean {
+    return !this.isLocked()
+  }
+
+  private saveAccount (pair: KeyringPair): Promise<KeyringPair$Json> {
+    this.addTimestamp(pair)
+    const json = pair.toJson(this._password)
+    return LocalStore.get(this._vaultKey).then((vault: any) => {
```

Comment:
why do we use `any` here but not other places

Reply:
point taken. It should be consistent